src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -592,6 +592,14 @@ private function addService($id, $definition)
             $return[] = sprintf("@throws InactiveScopeException when the '%s' service is requested while the '%s' scope is not active", $id, $scope);
         }
 
+        if ($definition->isDeprecated()) {
+            if ($return && 0 === strpos($return[count($return) - 1], '@return')) {
+                $return[] = '';
+            }
+
+            $return[] = '@deprecated';
```

Comment:
@nicolas-grekas the PhpStorm plugin is parsing service definitions, not the phpdoc of generated classes